Applications/Spire/Source/Ui/DropDownBox.cpp

```diff
@@ -0,0 +1,249 @@
+      set(Visibility::INVISIBLE);
+    style.get(Disabled() >> is_a<Icon>()).
+      set(Fill(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
+    return style;
+  }
+
+  auto ICON_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0, 0, 0, 0))).
+      set(Fill(QColor::fromRgb(0x33, 0x33, 0x33)));
+    return style;
+  }
+
+  auto TEXT_BOX_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(PaddingRight(scale_width(22)));
+    style.get(ReadOnly()).
+      set(PaddingRight(scale_width(0)));
+    return style;
+  }
+}
+
+class DropDownBox::DropDownListWrapper : public QWidget {
+  public:
+    DropDownListWrapper(ListView& list_view, QWidget* parent)
+        : QWidget(parent),
+          m_list_view(&list_view) {
+      m_drop_down_list = new DropDownList(*m_list_view, parent);
+      m_panel = m_drop_down_list->window();
+      m_drop_down_list->setFocusProxy(m_list_view);
+      auto list_model = m_list_view->get_list_model();
+      for(auto i = 0; i < list_model->get_size(); ++i) {
+        m_list_view->get_list_item(i)->setFocusPolicy(Qt::NoFocus);
+      }
+      m_list_view->installEventFilter(this);
+      m_panel->installEventFilter(this);
+    }
+
+    QWidget* get_panel() const {
+      return m_panel;
+    }
+
+  protected:
+    bool eventFilter(QObject* watched, QEvent* event) override {
+      if(watched == m_list_view) {
+        if(event->type() == QEvent::KeyPress) {
+          auto key_event = static_cast<QKeyEvent*>(event);
+          switch(key_event->key()) {
+            case Qt::Key_Tab:
+            case Qt::Key_Backtab:
+            case Qt::Key_Escape:
+              hide();
+              parentWidget()->setFocus();
+              QCoreApplication::sendEvent(parentWidget(), event);
+              break;
+          }
+        }
+      } else if(watched == m_panel) {
+        if(event->type() == QEvent::Close) {
+          hide();
+        } else if(event->type() == QEvent::KeyPress) {
+          QCoreApplication::sendEvent(m_list_view, event);
+        } else if(event->type() == QEvent::MouseButtonPress) {
+          auto mouse_event = static_cast<QMouseEvent*>(event);
+          if(parentWidget()->rect().contains(
+              parentWidget()->mapFromGlobal(mouse_event->globalPos()))) {
+            m_panel->setAttribute(Qt::WA_NoMouseReplay);
+          }
+        }
+      }
+      return QWidget::eventFilter(watched, event);
+    }
+
+    void showEvent(QShowEvent* event) override {
+      m_drop_down_list->show();
+      QWidget::showEvent(event);
+    }
+
+    void hideEvent(QHideEvent* event) override {
+      m_drop_down_list->hide();
+      QWidget::hideEvent(event);
+    }
+
+  private:
+    ListView* m_list_view;
+    DropDownList* m_drop_down_list;
+    QWidget* m_panel;
+};
+
+DropDownBox::DropDownBox(ListView& list_view, QWidget* parent)
+    : QWidget(parent),
+      m_list_view(&list_view),
+      m_submission(m_list_view->get_current_model()->get_current()) {
```

Comment:
Why is the submission set to the current value?

Reply:
It's like the TextBox which initializes the submission with the value of the current model. Here I think the the current model of the ListView is the current model of the DropDownBox, so the current value is used to initialize the submission.